Automattic/newspack-block-theme
# Feat: add post subtitle support

### All Submissions:






### Changes proposed in this Pull Request:



Adds post subtitle functionality from the Newspack Theme. The only difference in UX is that the subtitle here is editable here _in-situ_, aka WYSIWYG, instead of in the post attributes sidebar:

<img width="441" alt="image" src="https://github.com/user-attachments/assets/9e39aea2-ee2d-4562-a8ef-137ef76790e7">

### How to test the changes in this Pull Request:

1. Ensure some posts have the `newspack_post_subtitle` post meta set
2. Edit a post, observe the post subtitle can be edited below the post title, save changes
3. Observe the changes are reflected on the front-end, in the single post view
4. Observe the post subtitle is also present in the archive pages and the search results page

### Other information:




---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202269484355327
